File: algorithms/binary_search.py

def binary_search(val, array, start, end):
    if start > end:
        return "not found"
    mid = int((end+start)/2)
    if val == array[mid]:
        return "found"
    if val < array[mid]:
        return binary_search(val, array, start, mid-1)
    else:
        return binary_search(val, array, mid+1, end)


li = [1, 2, 3, 4, 5, 6, 7, 8, 9]
print(len(li)/2)

print(binary_search(10, li, 0, len(li)-1))

File: algorithms/binary_search.c
#include <stdio.h>

int binary_search(int *tab, int cle, int n)
{
    int trouve = 0;
    int start = 0;
    int end = n - 1;
    int m = (start + end) / 2;
    while (!trouve)
    {
        if (start > end)
        {
            return 0;
        }
        if (tab[m] == cle)
        {
            trouve = 1;
            return trouve;
        }
        else
        {
            if (tab[m] < cle)
            {
                start = m + 1;
                m = (start + end) / 2;
            }
            else
            {
                end = m - 1;
                m = (start + end) / 2;
            }
        }
    }
}

int main()
{
    int tab1[5] = {1, 4, 6, 8, 9};
    if (binary_search(tab1, 6, 5))
        printf("found");
    else
        printf("not found");
}